src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1021,6 +1035,33 @@ def upgrade_agents(cmd, client, resource_group_name, cluster_name, kube_config=N
                                 summary='Problem loading the helm existing user supplied values')
         raise CLIInternalError("Problem loading the helm existing user supplied values: " + str(e))
 
+    # Change registry_path to private registry if PCR is set in helm values
+    existing_user_values_flat = utils.flatten(existing_user_values)
+    helm_container_repository = existing_user_values_flat.get('systemDefaultValues.image.repository')
+    registry_username = None
+    registry_password = None
+    if helm_container_repository.split('.')[0] != "mcr":
```

Comment:
With the introduction of pcr, it becomes a little tricky to set user fault in case of helm upgrade cmd failures. I think it would make sense to mark all failures in 'helm upgrade' commands as user faults in pcr cases(Referring [here](https://github.com/Azure/azure-cli-extensions/blob/abd529901037630e61ef29df58b0f27c1ca6986d/src/connectedk8s/azext_connectedk8s/custom.py#L1061)).  Let me know if you can think of a scenario where is might not be a user fault.